### PR TITLE
KTOR-7812 Use default JDK for running tests on CI

### PR DESCRIPTION
**Pull request by osipxd**
**Subsystem**
Tests infrastructure

**Motivation**
[KTOR-7812](https://youtrack.jetbrains.com/issue/KTOR-7812) Use default JDK for running tests on CI

**Solution**
Use the JDK from `JAVA_HOME` to run tests on CI.

It is better to review commit by commit.

### CODE REVIEW

**Comment by bjhham** [comment]
There seems to be some problems in the TLS package - we got an out-dated ECDH curve `java.security.ProviderException: Curve not supported:  secp128r1 (1.3.132.0.28)` and some rogue reflection `java.lang.reflect.InaccessibleObjectException: Unable to make private static java.time.Instant java.time.Instant.create(long,int) accessible: module java.base does not "opens java.time" to unnamed module @5b8bc155`

You also seem to have unlocked a consistent failure in `testHeaderIsTooLong()`, so that's good 😄

**Comment by osipxd** [comment]
Yes, finally runs against different Java versions found something :)

File: ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/plugins/CookiesIntegrationTests.kt [hunk 1]
```diff
@@ -186,7 +186,7 @@ class CookiesIntegrationTests : ClientLoader() {
     }
 
     @Test
-    fun testCookiesWithWrongValue() = clientTests(listOf("js", "Darwin", "DarwinLegacy", "WinHttp")) {
+    fun testCookiesWithWrongValue() = clientTests(listOf("Js", "Darwin", "DarwinLegacy", "WinHttp", "Java")) {
```
Comment (osipxd) [inline ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/plugins/CookiesIntegrationTests.kt:189]: https://github.com/openjdk/jdk16/commit/1c47244b01dd9e7e176dd91215baef2afbc2626f: Java 16 adds validation for header values, so this test doesn't pass on Java engine anymore

**Comment by osipxd** [comment]
@bjhham, finally I've fixed this PR. After some tries I didn't manage to fix `testHeaderIsTooLong`, so I disabled it and created a task to fix it later:
- [KTOR-7811](https://youtrack.jetbrains.com/issue/KTOR-7811) Flaky test: SustainabilityTestSuite.testHeaderIsTooLong

**Comment by bjhham** [comment]
Looks like there's just one test being a nuisance here `io.ktor.tests.server.tomcat.jakarta.TomcatHttpServerCommonTest`.

Might need to ignore it idk.

**Review by bjhham — APPROVED** [review]

**Comment by e5l** [comment]
testHeadRequest requires investigation

**Comment by e5l** [comment]
@osipxd, could you put @Ignore on this test in your branch? I'm investigating it in the separate one

**Review by e5l — APPROVED** [review]
Please consider adding retry for failing tests on native

**Pull request merged**